### PR TITLE
 Trace changes Insurance from BackOffice

### PR DESCRIPTION
**Pull request by dkaplunov-fd**

### CODE REVIEW

File: src/main/java/com/hedvig/backoffice/services/members/MemberServiceStub.java [hunk 1]
```diff
@@ -167,4 +167,11 @@ public void editMember(String memberId, MemberDTO memberDTO, String token) {
         .filter(u -> ids.contains(u.getMemberId().toString()))
         .collect(Collectors.toList());
   }
+
+  @Override
+  public void setFraudulentStatus(String memberId, MemberFraudulentStatusDTO dto, String token) {
+
```
Comment (meletios) [inline src/main/java/com/hedvig/backoffice/services/members/MemberServiceStub.java]: we can just randomly set it here

File: src/main/java/com/hedvig/backoffice/services/product_pricing/ProductPricingServiceStub.java [hunk 2]
```diff
@@ -136,6 +134,10 @@ public void activate(String memberId, InsuranceActivateDTO dto, String token) {
   }
 
   @Override
+  public void cancel(String memberId, InsuranceCancellationDateDTO dto, String token) {
+
```
Comment (meletios) [inline src/main/java/com/hedvig/backoffice/services/product_pricing/ProductPricingServiceStub.java]: also here

File: src/main/java/com/hedvig/backoffice/web/MemberController.java [hunk 3]
```diff
@@ -141,7 +145,11 @@ public InsuranceStatusDTO insurance(
       @PathVariable String hid,
       @RequestBody InsuranceCancellationDTO dto,
       @AuthenticationPrincipal Principal principal) {
-    memberService.cancelInsurance(hid, dto, personnelService.getIdToken(principal.getName()));
+    InsuranceCancellationDateDTO sendDto = new InsuranceCancellationDateDTO(
+      dto.getMemberId(), dto.getInsuranceId(),
+      dto.getCancellationDate().atStartOfDay(ZoneId.of("Europe/Stockholm")).toInstant());
+//    memberService.cancelInsurance(hid, dto, personnelService.getIdToken(principal.getName()));
```
Comment (meletios) [inline src/main/java/com/hedvig/backoffice/web/MemberController.java]: remove comments

File: src/main/java/com/hedvig/backoffice/web/MemberController.java [hunk 4]
```diff
@@ -200,8 +208,15 @@ public InsuranceSearchResultDTO searchInsurancePaged(
...
   @PostMapping("/insurance/{memberId}/insuredAtOtherCompany")
   public ResponseEntity<?> setInsuredAtOtherCompany(
-      @PathVariable String memberId, @RequestBody @Valid InsuredAtOtherCompanyDTO dto) {
-    productPricingService.setInsuredAtOtherCompany(memberId, dto);
+    @PathVariable String memberId, @RequestBody @Valid InsuredAtOtherCompanyDTO dto, @AuthenticationPrincipal Principal principal) {
+    productPricingService.setInsuredAtOtherCompany(memberId, dto, personnelService.getIdToken(principal.getName()));
+    return ResponseEntity.noContent().build();
+  }
+
+  @PostMapping("/insurance/{memberId}/fraudulentStatus")
```
Comment (meletios) [inline src/main/java/com/hedvig/backoffice/web/MemberController.java]: it should be `/{memberId}/setFraudulentStatus`

File: src/main/java/com/hedvig/backoffice/services/members/MemberServiceStub.java (hunk 1 above)
Comment (dkaplunov-fd) [reply src/main/java/com/hedvig/backoffice/services/members/MemberServiceStub.java]: done

File: src/main/java/com/hedvig/backoffice/services/product_pricing/ProductPricingServiceStub.java (hunk 2 above)
Comment (dkaplunov-fd) [reply src/main/java/com/hedvig/backoffice/services/product_pricing/ProductPricingServiceStub.java]: done

File: src/main/java/com/hedvig/backoffice/web/MemberController.java (hunk 3 above)
Comment (dkaplunov-fd) [reply src/main/java/com/hedvig/backoffice/web/MemberController.java]: done

File: src/main/java/com/hedvig/backoffice/web/MemberController.java (hunk 4 above)
Comment (dkaplunov-fd) [reply src/main/java/com/hedvig/backoffice/web/MemberController.java]: done

**Pull request merged**